samples/ResourceOwnerPasswordFlow/Startup.cs

```diff
@@ -0,0 +1,256 @@
+                {
+                    //
+                    // serve the resource server
+                    //
+
+                    var authenticated = context.User.Identity.IsAuthenticated;
+                    var name = context.User.Identity.Name;
+
+                    var builder = new StringBuilder();
+
+                    builder.Append("<h2>General Info</h2>");
+
+                    builder.AppendFormat("<p>Time: {0}</p>", DateTime.Now.ToString());
+                    builder.AppendFormat("<p>Environment: {0}</p>", environment.RuntimeType);
+                    builder.AppendFormat("<p>IsAuthenticated: {0}</p>", context.User.Identity.IsAuthenticated);
+
+                    if (context.User.Identity.IsAuthenticated)
+                    {
+                        //
+                        // serve protected resources 
+                        // 
+
+                        builder.Append("<h2>Secure Resources</h2>");
+                        builder.AppendFormat("<p>User Name: {0}</p>", context.User.Identity.Name);
+
+                        builder.Append("<h2>User Claims</h2>");
+                        builder.Append("<dl>");
+                        foreach (var claim in context.User.Claims)
+                        {
+                            builder.AppendFormat("<dt>{0}</dt><dd>{1}</dd>", claim.Type, claim.Value);
+                        }
+                    }
+
+                    await context.Response.WriteAsync(builder.ToString());
+                }
+                else
+                {
+                    //
+                    // serve the relying party
+                    //
+
+                    context.Response.Redirect("relying-party.html");
+                }
+            });
+        }
+
+        private SigningCredentials CreateSigningCredentials()
+        {
+            var certificate = LoadCertificate();
+            var key = new X509SecurityKey(certificate);
+
+            var credentials = new SigningCredentials(key,
+                SecurityAlgorithms.RsaSha256Signature,
+                SecurityAlgorithms.Sha256Digest);
+
+            return credentials;
+        }
+
+        private X509Certificate2 LoadCertificate()
+        {
+            var resourceName = "ResourceOwnerPasswordFlow.Certificate.pfx";
+            using (var stream = this.GetType()
+                .Assembly
+                .GetManifestResourceStream(resourceName))
+
+            using (var buffer = new MemoryStream())
+            {
+                stream.CopyTo(buffer);
+                buffer.Flush();
+
+                // azure web apps require `MachineKeySet` for this to work
+                return new X509Certificate2(
+                    buffer.ToArray(),
+                    "Owin.Security.OpenIdConnect.Server",
+                    X509KeyStorageFlags.MachineKeySet);
+            }
+        }
+
+        private void CreateUsers(IServiceProvider serviceProvider)
```

Comment:
Done. I'm not sure how the `async` works to our advantage here. I just understand, at this point, that it works.